pkg/discovery/consul/consul.go

```diff
@@ -28,20 +28,23 @@ func (c *Consul) Resolver(address string) (discovery.Resolver, error) {
 	}
 
 	name := uri.Host
-
 	if watcher, ok := c.watchers[name]; ok {
 		return watcher, nil
 	}
 
+	// Create a new watcher
```

Comment:
new watcher?

Reply:
if there is no watcher in the cache already